aloha/real_env.py

```diff
@@ -241,38 +270,52 @@ def reset(self, fake=False):
             observation=self.get_observation(),
         )
 
-    def step(self, action, base_action=None, get_base_vel=False, get_obs=True):
-        state_len = int(len(action) / 2)
-        left_action = action[:state_len]
-        right_action = action[state_len:]
-        self.follower_bot_left.arm.set_joint_positions(left_action[:6], blocking=False)
-        self.follower_bot_right.arm.set_joint_positions(right_action[:6], blocking=False)
-        self.set_gripper_pose(left_action[-1], right_action[-1])
+    def step(self, action, base_action=None, get_obs=True):
+
+        # Dynamically calculate per-bot state length
+        state_len = int(len(action) / len(self.follower_robots))
+        index = 0
+
+        # Iterate through each follower bot and set joint positions
+        for name, robot in self.follower_robots.items():
+            bot_action = action[index:index + state_len]
+            robot.arm.set_joint_positions(
+                bot_action[:6], blocking=False)  # Set arm positions
+            self.set_gripper_pose(name, bot_action[-1])  # Set gripper position
+            index += state_len
+
         if base_action is not None:
             base_action_linear, base_action_angular = base_action
-            self.base.base.command_velocity_xyaw(x=base_action_linear, yaw=base_action_angular)
-        if get_obs:
-            obs = self.get_observation(get_base_vel)
-        else:
-            obs = None
+            self.base.base.command_velocity_xyaw(
+                x=base_action_linear, yaw=base_action_angular)
+
+        # Optionally get observations
+        obs = self.get_observation() if get_obs else None
+
         return dm_env.TimeStep(
             step_type=dm_env.StepType.MID,
             reward=self.get_reward(),
             discount=None,
-            observation=obs)
+            observation=obs,
+        )
 
 
-def get_action(
-    leader_bot_left: InterbotixManipulatorXS,
-    leader_bot_right: InterbotixManipulatorXS
-):
-    action = np.zeros(14)  # 6 joint + 1 gripper, for two arms
-    # Arm actions
-    action[:6] = leader_bot_left.core.joint_states.position[:6]
-    action[7:7+6] = leader_bot_right.core.joint_states.position[:6]
-    # Gripper actions
-    action[6] = LEADER_GRIPPER_JOINT_NORMALIZE_FN(leader_bot_left.core.joint_states.position[6])
-    action[7+6] = LEADER_GRIPPER_JOINT_NORMALIZE_FN(leader_bot_right.core.joint_states.position[6])
+def get_action(robots: dict):
```

Comment:
More precise typing of dicts

```suggestion
def get_action(robots: dict[str, InterbotixManipulatorXS]):
```